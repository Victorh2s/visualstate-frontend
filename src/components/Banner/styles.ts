import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`

  width: 100%;
  height: 22rem;
  overflow: hidden;
  cursor: pointer;


  `}
`;

export const Banner = styled.img`
  ${({ theme }) => css`
  width: 100%;

  `}
`;

export const GladientVertical = styled.div`
  ${({ theme }) => css`


  width: inherit;
  height: inherit;
  background: linear-gradient(to bottom, #0d0c0c 5%, transparent 100%);


  `}
`;

export const DivButtons = styled.div`
  ${({ theme }) => css`

  margin-top: -3rem;
  float: right;
  width: 10%;
  height:3rem;
  border-top-left-radius: 10px;
  background: ${theme.colors.dark};
  text-align: center;
  cursor: pointer;

  > svg {
    color: ${theme.colors.white};
  `}
`;

export const Edit = styled.button`
  ${({ theme }) => css`
  border: none;
  width: 3rem;
  height:3rem;
  background: transparent;
  cursor: pointer;

  > svg {
    color: ${theme.colors.white};
  `}
`;
