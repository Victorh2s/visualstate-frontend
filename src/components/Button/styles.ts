import styled, { css } from 'styled-components';
import { ButtonProps } from '.';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background: ${theme.colors[color]};
    border: none;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    cursor: pointer;
    border-radius: 1rem;
    transition: all 300ms ease-in-out;
    display: flex;
    align-items: center;
    justify-content: center;

    &:focus {
      outline: none;
      box-shadow: 0 0 ${theme.spacings.small} ${theme.colors[color]};
      filter: brightness(110%);
    }

    &:hover {
      filter: brightness(90%);
    }

    &:disabled {
      background: ${theme.colors.gray8};
      color: ${theme.colors.white};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 2rem;
      height: 2rem;
      margin-left: 1rem;
    }
  `}
`;
