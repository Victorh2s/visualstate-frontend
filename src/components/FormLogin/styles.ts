import styled, { css } from 'styled-components';

export const Form = styled.form`
`;

export const ButtonWrapper = styled.div``;

export const Container = styled.div`
    text-align: center;
    width: 35vw;
`;

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: 100%;
    margin-bottom: ${theme.spacings.large};

  `}
`;

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    position: relative;

    > svg {
      position: absolute;
      top: 50%;
      transform: translate(0, -50%);
      right: 1rem;
      width: 2.5rem;
      height: 2.5rem;
      color: ${theme.colors.gray6};
      z-index: ${theme.layers.layer1};

    }

    *:focus ~ svg {
      color: ${theme.colors.primary};

    }

  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    border: 1px solid ${theme.colors.gray3};
    width: 100%;
    height: 100%;
    font-size: ${theme.font.sizes.normal};
    transition: ${theme.transitions.fast};
    padding: ${theme.spacings.small} ${theme.spacings.xsmall};
    background: ${theme.colors.white};
    padding-right: 3.7rem;
    border-radius: ${theme.spacings.tiny};
    outline: none;

    &::placeholder {
      visibility: hidden;
      opacity: 0;
    }

    &:focus {
      border: ${theme.spacings.xxtiny} solid ${theme.colors.primary};
      box-shadow: 0 0 ${theme.spacings.xtiny} 0 ${theme.colors.primary};
      background: ${theme.colors.white};
    }

    &:focus
      + ${Label},
      &:not(:placeholder-shown)
      + ${Label},
      &:-webkit-autofill
      + ${Label} {
      top: 0;
      transform: translate(0, -50%);
      font-size: ${theme.font.sizes.xsmall};
      color: ${theme.colors.white};
      background: ${theme.colors.primary};
      filter: none;
    }


  `}
`;

export const Label = styled.label<{ element: string }>`
  ${({ theme, element }) => css`
    display: inline;
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    left: ${theme.spacings.xsmall};
    font-size: ${theme.font.sizes.small};
    height: ${theme.font.sizes.small};
    transition: ${theme.transitions.fast};
    background: ${theme.colors.white};
    padding: 0.5rem 1rem;
    display: flex;
    align-items: center;
    line-height: 1;
    z-index: ${theme.layers.layer1};
    color: ${theme.colors.gray6};
    border-radius: ${theme.spacings.tiny};

    ${
      element === 'textarea' &&
      css`
      top: ${theme.spacings.large};
      transform: translate(0, -50%);
    `
    }
  `}
`;

export const DivButton = styled.div`

`;
export const DivLink = styled.div`
  ${({ theme }) => css`

    margin-bottom: ${theme.spacings.large};
    > a{
      text-decoration: none;
    }
  `}
`;

export const button = styled.button`
  ${({ theme }) => css`

    background: ${theme.colors.primary};
    border: none;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    cursor: pointer;
    border-radius: 1rem;
    transition: all 300ms ease-in-out;
    display: flex;
    margin: 0 auto;

    &:focus {
      outline: none;
      box-shadow: 0 0 ${theme.spacings.small} ${theme.colors.primary};
      filter: brightness(110%);
    }

    &:hover {
      filter: brightness(90%);
    }



    > svg {
      width: 2rem;
      height: 2rem;
      margin-left: 1rem;
    }
  `}
`;

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.warning};
    margin: ${theme.spacings.xxsmall} 0 ${theme.spacings.medium};
  `}
`;
