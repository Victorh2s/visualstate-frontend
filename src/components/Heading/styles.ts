import styled, { css, DefaultTheme } from 'styled-components';
import { HeadingProps } from '.';

const titleSize = {
  supersmall: (theme: DefaultTheme) => css`
  font-size: ${theme.font.sizes.xxsmall};
    ${mediaFont(theme)};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
    ${mediaFont(theme)};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${mediaFont(theme)};
  `,
  big: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
    ${mediaFont(theme)};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
    ${mediaFont(theme)};
  `,
};

const mediaFont = (theme: DefaultTheme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.medium};
  }
  @media ${theme.media.lteSmall} {
    font-size: ${theme.font.sizes.small};
  }
  @media ${theme.media.lteSuperSmall} {
    font-size: ${theme.font.sizes.xxsmall};
  }
`;

const titleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1<HeadingProps>`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.dark : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
