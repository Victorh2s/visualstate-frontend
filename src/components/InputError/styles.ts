import styled, { css } from 'styled-components';

export const ErrorMessage = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.warning};
    margin: ${theme.spacings.xxsmall} 0 ${theme.spacings.large};
  `}
`;
