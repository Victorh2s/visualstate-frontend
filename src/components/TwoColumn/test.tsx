import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { TwoColumn, TwoColumnProps } from '.';

const props: TwoColumnProps = {
  text: 'Texto',
  srcImg: 'https://source.unsplash.com/random/1000x1000?r=1',
};

describe('<TwoColumn />', () => {
  it('should render TwoColumn', () => {
    renderTheme(<TwoColumn {...props} />);

    expect(screen.getByRole('heading', { name: 'Texto' })).toBeInTheDocument();
    expect(screen.getByLabelText('Video')).toBeInTheDocument();
  });

  it('should match to snapshot', () => {
    const { container } = renderTheme(<TwoColumn {...props} />);

    expect(container).toMatchSnapshot();
  });
});
