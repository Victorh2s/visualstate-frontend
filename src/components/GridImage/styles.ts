import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
${({ theme }) => css`
padding: 2rem;
padding-bottom: 10rem;
 > ${TextComponent}{
  text-align: center;
  margin-bottom: ${theme.spacings.xhuge}
}
${HeadingContainer}{
  text-align: center;
}
`}
`;

export const Grid = styled.div`
${({ theme }) => css`
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px, 1fr));
  gap: ${theme.spacings.large}
`}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;
  &:hover{
    transform: scale(1.2);
  }
`;
