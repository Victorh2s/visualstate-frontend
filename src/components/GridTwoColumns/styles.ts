import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export type translatey = {
  translatey: number;
};

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    overflow-x: hidden;
    align-items: center;
    width: 100%;
    gap: ${theme.spacings.large};
    @media ${theme.media.lteOrEqMedium} {
      display: block;
      text-align: center;
      align-items: center;
      overflow-x: hidden;
    }
    @media ${theme.media.lteOrEqSmallest} {
      width: 70%;
  }
    ${Title} {
      margin-bottom: 0;
    }
  `}
`;

export const TextContainer = styled.div`
    width: 35rem;
    display: flex;
    overflow-x: hidden;
    margin: 0 auto;
    padding-bottom: 0.1rem;
`;

export const Ytext = styled.div<translatey>`
  ${({ theme, translatey }) => css`
    display: flex;
    transform: translateX(-${translatey}rem);
    transition: all 400ms ease-in-out;
    padding-bottom: 0.1rem;
    @media ${theme.media.lteOrEqSmallest} {
      transform: translateX(-${translatey - 11}rem);
      transition: all 400ms ease-in-out;
    }
`}
`;

export const HeadingHeight = styled.div`
 ${({ theme }) => css`
  width: 35rem;
  padding-bottom: 0.1rem;
  @media ${theme.media.lteOrEqSmallest} {
      width: 25rem;
  }
  `}
`;

export const Carrousel = styled.div`
 ${({ theme }) => css`
  height: 40rem;
  margin: 0 auto;
  overflow-y: hidden;
  @media ${theme.media.lteOrEqSmallest} {
    height: 25rem;
  }

  `}
`;

export const Yvideo = styled.div<translatey>`
  ${({ theme, translatey }) => css`
    transform: translateY(-${translatey}rem);
    transition: all 400ms ease-in-out;
    @media ${theme.media.lteOrEqMedium} {
      transform: translateY(-${translatey + 12}rem);
      transition: all 400ms ease-in-out;
  }

  `}
`;

export const SourceContainer = styled.div`
  width: 100%;
  height: 50rem;
`;

export const videos = styled.video`
  ${({ theme }) => css`
  width: 45vw;
  height: 50rem;
  @media ${theme.media.lteOrEqMedium} {
    width: 80%;
    height: 50rem;
    align-items: center;
  }
  @media ${theme.media.lteOrEqSmallest} {
    width: 60%;
    height: 50rem;
    align-items: center;
  }
    `}
`;
