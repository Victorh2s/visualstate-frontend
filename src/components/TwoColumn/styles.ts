import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Carousel = styled.div`
`;

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    min-width: 90vw;
    padding: ${theme.spacings.small};
    border: 0.1rem solid white;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};
    @media ${theme.media.lteOrEqMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }
    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteOrEqMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const SourceContainer = styled.div`

`;

export const video = styled.video`
width: 100%;
`;
