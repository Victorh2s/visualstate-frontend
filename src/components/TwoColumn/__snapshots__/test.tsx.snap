// Vitest Snapshot v1

exports[`<TwoColumn /> > should match to snapshot 1`] = `
.c3 {
  color: #FFFFFF;
  font-size: 3.2rem;
  text-transform: uppercase;
}

.c0 {
  display: grid;
  min-width: 90vw;
  padding: 1.6rem;
  border: 0.1rem solid white;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c0 .c2 {
  margin-bottom: 4.0rem;
}

.c4 {
  width: 100%;
}

@media (max-width:768px) {
  .c3 {
    font-size: 2.4rem;
  }
}

@media (max-width:414px) {
  .c3 {
    font-size: 1.6rem;
  }
}

@media (max-width:362px) {
  .c3 {
    font-size: 1.2rem;
  }
}

@media (max-width:768px) {
  .c0 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c1 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2 c3"
      >
        Texto
      </h2>
    </div>
    <div
      class=""
    >
      <video
        aria-label="Video"
        autoplay=""
        class="c4"
        loop=""
        src="https://source.unsplash.com/random/1000x1000?r=1"
      />
    </div>
  </div>
</div>
`;
